src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,296 @@
+
+import org.apache.commons.lang3.Validate;
+import javax.annotation.Nullable;
+import java.util.*;
+
+/**
+ * @author Emerson
+ * @since 12/27/2020
+ * @author rycbar
+ * @since 22.09.2022
+ *
+ */
+public final class LitematicaSchematic extends StaticSchematic {
+    int minX=0;
+    int minY=0;
+    int minZ=0;
+    private static final String reg = "Regions";
+    private static final String meta = "Metadata";
+    private static final String schemSize = "EnclosingSize";
+    private static final String blSt = "BlockStates";
+    private static final String blStPl = "BlockStatePalette";
+    private static final String pos = "Position";
+    private static final String size = "Size";
+
+    public LitematicaSchematic(NBTTagCompound nbt) {
+        getMinimumCorner(nbt);
+
+        this.x = Math.abs(nbt.getCompoundTag(meta).getCompoundTag(schemSize).getInteger("x"));
+        this.y = Math.abs(nbt.getCompoundTag(meta).getCompoundTag(schemSize).getInteger("y"));
+        this.z = Math.abs(nbt.getCompoundTag(meta).getCompoundTag(schemSize).getInteger("z"));
+        this.states = new IBlockState[this.x][this.z][this.y];
+
+        for (String subReg : getRegions(nbt)) {
+            NBTTagList usedBlockTypes = nbt.getCompoundTag(reg).getCompoundTag(subReg).getTagList(blStPl, 10);
+            IBlockState[] blockList = getBlockList(usedBlockTypes);
+
+            int bitsPerBlock = getBitsPerBlock(usedBlockTypes.tagCount());
+            long regionVolume = getVolume(nbt, subReg);
+            long[] blockStateArray = getBlockStates(nbt, subReg);
+
+            LitematicaBitArray bitArray = new LitematicaBitArray(bitsPerBlock, regionVolume, blockStateArray);
+
+            writeSubregionIntoSchematic(nbt, subReg, blockList, bitArray);
+        }
+    }
+
+    /**
+     * @return Array of subregion names.
+     */
+    private static String[] getRegions(NBTTagCompound nbt) {
+        return nbt.getCompoundTag(reg).getKeySet().toArray(new String[0]);
+    }
+
+    /**
+     * Calculates the minimum cords/origin of the schematic as litematica schematics
+     * can have a non-minimum origin.
+     */
+    private void getMinimumCorner(NBTTagCompound nbt) {
+        for (String subReg : getRegions(nbt)) {
+            this.minX = Math.min(this.minX, getMinimumCord(nbt, subReg,"x"));
+            this.minY = Math.min(this.minY, getMinimumCord(nbt, subReg,"y"));
+            this.minZ = Math.min(this.minZ, getMinimumCord(nbt, subReg,"z"));
+        }
+    }
+
+    /**
+     * @param s axis that should be read.
+     * @return the lower cord of the requested axis.
+     */
+    private static int getMinimumCord(NBTTagCompound nbt, String subReg, String s) {
+        int a = nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(pos).getInteger(s);
+        int b = nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(size).getInteger(s);
+        if (b < 0) {
+            b++;
+        }
+        return Math.min(a,a+b);
```

Comment:
what is going on here? can size be negative????

Reply:
litematicas schematics can point in all 8 cubical quadrants and therefor can have boxes with negative coordinat sizes.
schematic origin can be anywhere inside our outside of the schematic box. (only if manuay changed during creation of schematic else origin will be minimum corner)
attached excample schematic i used for testing (as txt because of github just remove the txt at the end)

[Main.litematic.txt](https://github.com/cabaletta/baritone/files/9677835/Main.litematic.txt)
[Main-cleartext.txt](https://github.com/cabaletta/baritone/files/9677847/Main-cleartext.txt)